hasher-matcher-actioner/hmalib/lambdas/stream/ddb_stream_processor.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+from hmalib.common.count_models import MatchByPrivacyGroupCounter
+import os
+import typing as t
+from collections import defaultdict
+import boto3
+
+from hmalib.common.logging import get_logger
+
+DYNAMODB_TABLE = os.environ["DYNAMODB_TABLE"]
+
+logger = get_logger(__name__)
+dynamodb = boto3.resource("dynamodb")
+
+
+def _is_hash_record(item_image: t.Dict) -> bool:
+    """
+    Returns True if the new image looks like a hash record. At present, supports
+    PDQ shapes only.
+    """
+    return (
+        "ContentHash" in item_image
+        and "HashType" in item_image
+        and item_image["HashType"] == {"S": "pdq"}
+        and "Quality" in item_image
+        and "SignalSource" not in item_image
+    )
+
+
+def _is_match_record(item_image: t.Dict) -> bool:
+    """
+    Returns True if the new image looks like a match record. At present, supports
+    PDQ shapes only.
+    """
+    result = (
+        "ContentHash" in item_image
+        and "HashType" in item_image
+        and item_image["HashType"] == {"S": "pdq"}
+        and "SignalSource" in item_image
+        and "Quality" not in item_image
+    )
+    return result
+
+
+def _get_keys_for_pg(pg: str) -> t.Tuple[str, str]:
+    """
+    Return partition and sort keys for a given privacy group.
+    """
+
+
+def lambda_handler(event, context):
+    """
+    Listens to events on the dynamodb table and increments various counters.
+
+    Presently supported:
+    ---
+    1. split of hash and matches by privacy group
+
+    Planned
+    ---
+    1. move total counters here
```

Comment:
[Diff 4/5] Also add TODO note where total counters are currently implemented stating that they are going to be moved (to avoid someone copying that pattern in the meantime)